compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java

```diff
@@ -610,6 +618,61 @@ public void visit(BLangInvocation invocationExpr) {
         }
     }
 
+    private boolean isFieldsInitializedForSelfKeyWordExpression(BLangInvocation invocationExpr) {
+
+        if (invocationExpr.expr == null || !isSelfKeyWordExpr(invocationExpr.expr)) {
+            return true;
+        }
+        StringBuilder uninitializedFields =
+                getUninitializedFieldsForSelfKeyword((BObjectType) ((BLangSimpleVarRef)
+                        invocationExpr.expr).symbol.type);
+        if (uninitializedFields.length() != 0) {
+            this.dlog.error(invocationExpr.pos, DiagnosticCode.CONTAINS_UNINITIALIZED_FIELDS,
+                    uninitializedFields.toString());
+            return false;
+        }
+        return true;
+    }
+
+    private boolean isFieldsInitializedForSelfKeyWordMethodInvocation(BLangInvocation invocationExpr) {
+
+        for (BLangExpression expr : invocationExpr.requiredArgs) {
```

Comment:
This only handles required args and named args right? Rest args also need to be handled. 

Shall we add tests too?

Reply:
Added.